subprojects/execution/src/main/java/org/gradle/internal/execution/caching/CachingDisabledReasonCatwgory.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.caching;
+
+public enum CachingDisabledReasonCatwgory {
+    /**
+     * Reason for disabled caching is not known.
+     */
+    UNKNOWN,
+
+    /**
+     * Caching has not been enabled for the build.
+     */
+    BUILD_CACHE_DISABLED,
+
+    /**
+     * Caching has not been enabled for the work.
+     */
+    NOT_CACHEABLE,
+
+    /**
+     * Condition enabling caching isn't satisfied.
+     */
+    ENABLE_CONDITION_NOT_SATISFIED,
+
+    /**
+     * Condition disabling caching satisfied.
+     */
+    DISABLE_CONDITION_SATISFIED,
+
+    /**
+     * The work has no outputs declared.
+     */
+    NO_OUTPUTS_DECLARED,
+
+    /**
+     * Work has declared output that is not cacheable.
+     */
+    NON_CACHEABLE_OUTPUT,
+
+    /**
+     * Work's outputs overlap with other work's.
+     */
+    OVERLAPPING_OUTPUTS,
+
+    /**
+     * The work's implementation is not cacheable.
+     *
+     * Reasons for non-cacheable implementations:
+     * <ul>
+     *     <li>the type is loaded via an unknown classloader,</li>
+     *     <li>a Java lambda was used (see https://github.com/gradle/gradle/issues/5510).</li>
+     * </ul>
+     */
+    // TODO Link to user guide instead of issue
```

Comment:
Will you fix those todos as part of this PR?

Reply:
Do we already have something to link to?

Reply:
Well, we have a warning in here: https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:how_does_it_work

Reply:
Added a link.

Reply:
There is a third TODO left for non-cacheable inputs.

Reply:
Fixed.